docs: network and configuration docs update for 1.14 (#1803)
software-mansion/radon-ide: packages/docs/docs/guides/configuration.md

When building the IDE, our approach was always to try and deal with different configurations automatically, without user input.
However, due to the spectrum of possible modifications to React Native project setups, it is very difficult to do that in some cases.
We constantly look into ways of improving things, so if the IDE doesn't work for your setup, please report an issue with a sample project that reproduces the setup that you use.
While some project setups (like brownfield projects) are more challenging than others, we want the IDE to be universally available for all types of React Native projects and will investigate the possibilities of expanding configuration options or handling some project modifications in an automated way.

## Managing configuration file

Radon IDE uses the standard VS Code `launch.json` format for customizing the build process to your project.
Before you can change any of the options you'll need to create a launch configuration file, unless you already have one in your project.

> ### Creating and updating `launch.js` using Radon IDE's panel

Reviewer: ```suggestion
### Creating and updating `launch.json` using Radon IDE's panel
```

Author: Fixed in a0035d6e

In order to create a new project configuration, open `Launch Configuration Selector` and select `Add custom launch config option` You will be presented with a wizard, that will allow you to configure the specific options in a GUI environment.

<img width="400" src="/img/docs/ide_add_custom_launch_config.png" className="shadow-image"/>

> The wizard provides simple means to generate and edit `launch.js` file and currently supports setting the basic launch options. If needed, the `Edit in launch.json` button can be used to jump to the file itself, which may then be edited manually.

Reviewer: ```suggestion
The wizard provides simple means to generate and edit the `launch.json` file.
If necessary, the `Edit in launch.json` button can be used to jump to the file itself, which may then be edited manually.
```

Author: Changed in a0035d6e
```md
The wizard provides simple means to generate and edit `launch.json`. If necessary, you can use the `Edit in launch.json` button to jump to the file itself and make changes manually.  
```

> After defining the set of options, `Save` button allows for creating the `launch.json` configuration file, which can then be used from the `Launch Configuration Selector`.

Reviewer: ```suggestion
After filling out the options, use the `Save` button to create the `launch.json` configuration file (if it does not exist yet) and write your launch configuration to it, which can then be selected from the `Launch Configuration Selector`.
```

> The file should be added to version control (git) as it carries configuration that is specific to your project setup rather than user specific editor settings.

Author: Changed in a0035d6e:
```md
You should consider adding the `launch.json` file to your project's version control system (e.g. `git`) as it carries configuration that is specific to your project setup rather than user specific editor settings.  
```

<img width="400" src="/img/docs/ide_launch_config_window.png" className="shadow-image"/>

> The existing custom launch configuration can be edited or deleted by clicking on the cog icon next to its name in `Launch Configuration Selector`.

Reviewer: ```suggestion
An existing custom launch configuration can be edited or deleted by clicking on the cog icon next to its name in `Launch Configuration Selector`.
```

}
```

## Other launch configuration customization

### Preview settings

The `preview` object allows you to customize the behavior of device preview:

| Attribute            | Description                                                                                                                                                   |
| -------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| **waitForAppLaunch** | Defaults to `true`, this should only be set to `false` for brownfield setups when React Native components aren't rendered immediately after the app launches. |

> 

Reviewer: ```suggestion

```
Pretty sure the option is not supported anymore 😔

Author: Removed in a0035d6e
